test/e2e/util_gatewayapi_test.go

```diff
@@ -103,19 +103,51 @@ func assertCrdExists(t *testing.T, crdname string) (string, error) {
 	return crdVersion, err
 }
 
+// deleteExistingCRDs deletes if the CRD of the given name exists and returns an error if not.
+func deleteExistingCRDs(t *testing.T, crdname string) error {
+	t.Helper()
+	crd := &apiextensionsv1.CustomResourceDefinition{}
+	name := types.NamespacedName{Namespace: "", Name: crdname}
+
+	err := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 30*time.Second, false, func(context context.Context) (bool, error) {
+		if err := kclient.Get(context, name, crd); err != nil {
+			t.Logf("failed to get crd %s: %v", name, err)
+			return false, nil
+		}
+		crdConditions := crd.Status.Conditions
+		for _, c := range crdConditions {
+			if c.Type == apiextensionsv1.Established && c.Status == apiextensionsv1.ConditionTrue {
+				return true, nil
+			}
+		}
+		t.Logf("failed to find crd %s to be Established", name)
```

Comment:
Is there a reason we need to check if the CRD is `Established` before deleting it?

Reply:
I added it for error handling if you think we can remove it.

Reply:
Hmm, yes it's extra error handling, but it seems to me it's already been confirmed as established, and from the perspective of this helper function's role to delete the CRD, I don't think it matters if it's established.

I'd vote to remove the the logic to keep things simple, and aligned with the roles of the functions (`deleteExistingCRD` just deletes, and `assertCrdExists` does the check for established). I'll leave it up to you though, its not a big concern to me.